Std/Data/List/Lemmas.lean

```diff
@@ -2185,6 +2187,22 @@ theorem disjoint_take_drop : ∀ {l : List α}, l.Nodup → m ≤ n → Disjoint
       refine ⟨fun h => h₀ _ (mem_of_mem_drop h) rfl, ?_⟩
       exact disjoint_take_drop h₁ (Nat.le_of_succ_le_succ h)
 
+/-! ### isPrefixOf and isSuffixOf -/
+
+theorem length_le_of_isPrefixOf [BEq α] {l₁ l₂ : List α} (h : l₁.isPrefixOf l₂) :
+    l₁.length ≤ l₂.length := by
+  match l₁, l₂ with
+  | [],    _     => simp
+  | _::_,  []    => simp [isPrefixOf] at h
+  | a::as, b::bs =>
+    simp [isPrefixOf] at h
+    simp [Nat.succ_le_succ_iff]
+    exact length_le_of_isPrefixOf h.2
```

Comment:
Non-terminal `simp`. Can you use `simpa`? Or `simp only`?

Reply:
I used `simpa` to avoid a non-terminal `simp`.